### ISSUE TITLE
Make it easy to run batch prediction script

### ISSUE THREAD

**Issue by tscholak** [ElementAI/picard#17]
Related to #16. We require a convenient way to generate predictions in batches that can be evaluated with the official Spider and CoSQL evaluation scripts.

**Comment by majorminus66** [comment]
Hi Torsten, another question -

**Comment by kotalaraghava** [comment]
Hey @tscholak,

I find this issue quite interesting and would like to contribute.

I am looking at [spider](https://github.com/taoyds/spider/tree/master/evaluation_examples), basically, we need to generate output in this formate [pred_example.txt](https://github.com/taoyds/spider/blob/master/evaluation_examples/pred_example.txt). 

I tried finding cosql metrics to see how the evaluation is done but I couldn't find any code.

and I see [cosql.py](https://github.com/ElementAI/picard/blob/main/seq2seq/metrics/cosql/cosql.py) and [spider.py](https://github.com/ElementAI/picard/blob/main/seq2seq/metrics/spider/spider.py) exactly same, maybe adding IM would be efficient ?

Let me know your thoughts, then I will start working on it.

**Comment by tscholak** [comment]
Hi @kotalaraghava,
It's great that you want to work on this! However, this issue is mostly solved, because there is [codalab_seq2seq.py](https://github.com/ElementAI/picard/blob/main/seq2seq/codalab_seq2seq.py) which will produce these files for both spider and cosql. What's missing is a convenient way to run this with `make`. If you were to pick up from where I left of, that would be great 👍 The first step for you would be to familiarize yourself with the script.

**Comment by kotalaraghava** [comment]
Hey @tscholak ,

I am looking at [codalab_seq2seq.py](https://github.com/ElementAI/picard/blob/main/seq2seq/codalab_seq2seq.py) to understand how to create `codalab.json` and execute the script.

I found `type error Text2SQLInput` https://github.com/ElementAI/picard/blob/8eba4020f334891a4c5bb83f960aa6b7af6db4e1/seq2seq/codalab_seq2seq.py#L139-L145 here,
pipe is expecting `Text2SQLInput` object but we are passing just keyword args, so I am assuming it is a bug. I am gonna fix by passing query, db_id as a `Text2SQLInput`  object and removing other arguments as we are already passing other arguments for pre-training model auto-config. and call this through make config file on eval image. 

Basically, I am assuming to create `codalab.json` and execute the script for spider, cosql by passing db_path, dev.json files correspondingly and generate prediction_sq.txt. 

And for batch prediction, I am gonna create dataset class for spdier, cosql and pass them through data loader then to the pipeline.

I hope I am on the right path.

Thanks,
Raghav.

**Comment by tscholak** [comment]
Hi,
That sounds great!

> I am gonna create dataset class for spdier, cosql and pass them through data loader then to the pipeline.

Using a dataset class here is optional. For spider, for instance, you can also point the script to a file with question/db id pairs. This makes it possible to use the script to submit models to the Spider leaderboard.

If you want to use a data loader, though, you don't have to write a new one. Use the ones that are used in the training/evaluation script.

**Comment by tscholak** [comment]
closed by #76.
